extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedTracer.java

```diff
@@ -5,54 +5,317 @@
 
 package io.opentelemetry.extension.incubator.trace;
 
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER;
+import static io.opentelemetry.api.trace.SpanKind.SERVER;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
 import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
 import io.opentelemetry.context.Scope;
-import java.util.concurrent.Callable;
+import java.util.Map;
+import java.util.function.BiConsumer;
 
-/** Provides easy mechanisms for wrapping standard Java constructs with an OpenTelemetry Span. */
+/**
+ * Utility class to simplify tracing.
+ *
+ * <p>The <a
+ * href="https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/extended-tracer/README.md">README</a>
+ * explains the use cases in more detail.
+ */
 public final class ExtendedTracer implements Tracer {
 
+  private final OpenTelemetry openTelemetry;
+
   private final Tracer delegate;
 
-  /** Create a new {@link ExtendedTracer} that wraps the provided Tracer. */
-  public static ExtendedTracer create(Tracer delegate) {
```

Comment:
This is the preferred pattern for creating instances in this repo, versus creating constructors with different overloads. The public static methods allow us to assign names that assist in conveying the intent / difference when multiple methods for creating exist.